lib/v2/screens/app/app.dart

```diff
@@ -0,0 +1,122 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:provider/provider.dart';
+import 'package:seeds/v2/blocs/authentication/viewmodels/bloc.dart';
+import 'package:seeds/v2/components/notification_badge.dart';
+import 'package:seeds/v2/constants/app_colors.dart';
+import 'package:seeds/i18n/widgets.i18n.dart';
+import 'package:seeds/providers/notifiers/connection_notifier.dart';
+import 'package:seeds/screens/app/ecosystem/ecosystem.dart';
+import 'package:seeds/v2/screens/profile_screens/profile/profile_screen.dart';
+import 'package:seeds/screens/app/wallet/wallet.dart';
+import 'package:seeds/v2/screens/app/interactor/viewmodels/bloc.dart';
+
+class App extends StatefulWidget {
+  const App();
+
+  @override
+  _AppState createState() => _AppState();
+}
+
+class _AppState extends State<App> with WidgetsBindingObserver {
+  final List<AppScreenItem> appScreenItems = [
+    AppScreenItem(
+      title: "Wallet".i18n,
+      icon: 'assets/images/navigation_bar/wallet.svg',
+      iconSelected: 'assets/images/navigation_bar/wallet_selected.svg',
+      screen: Wallet(),
+      index: 0,
+    ),
+    AppScreenItem(
+      title: "Explore".i18n,
+      icon: 'assets/images/navigation_bar/explore.svg',
+      iconSelected: 'assets/images/navigation_bar/explore_selected.svg',
+      screen: Ecosystem(),
+      index: 1,
+    ),
+    AppScreenItem(
+      title: "Profile".i18n,
+      icon: 'assets/images/navigation_bar/user_profile.svg',
+      iconSelected: 'assets/images/navigation_bar/user_profile_selected.svg',
+      screen: ProfileScreen(),
+      index: 2,
+    ),
+  ];
+  PageController pageController = PageController(initialPage: 0, keepPage: true);
+
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance!.addObserver(this);
+  }
+
+  @override
+  void didChangeAppLifecycleState(AppLifecycleState state) {
+    switch (state) {
+      case AppLifecycleState.inactive:
+        break;
+      case AppLifecycleState.paused:
+        break;
+      case AppLifecycleState.resumed:
+        // Fires again auth on app resume
+        BlocProvider.of<AuthenticationBloc>(context).add(const InitAuthStatus());
+        Provider.of<ConnectionNotifier>(context, listen: false).discoverEndpoints();
+        break;
+      case AppLifecycleState.detached:
+        break;
+    }
+  }
+
+  @override
+  void dispose() {
+    WidgetsBinding.instance!.removeObserver(this);
```

Comment:
`!` can be replace by `?`

Reply:
As a rule of thumb using `!` is a red flag, It means something is probably wrong. Only in rare cases we should use `!`

Reply:
I copy this from the docs they manage it with !, Should I use ?

Reply:
yes, use  ?